### PR TITLE
ENH: Range widget

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

Add a widget to manipulate the Range comparison dataclass while providing a helpful visualization of the range.
The visualization will change its colors and update the relative spacing of the thresholds as the user changes them.

I've implemented everything I could think of for the visualization while keeping it simple. Some ideas that I discarded because they sounded annoying:
- logarithmic ranges
- single QPainter instead of multi widgets in a grid
- use the mathematical filled/empty dots on the bounds to show inclusive/not

## Motivation and Context

Continues #30 
The visualization is motivated by a need to disambiguate what exactly the behavior of the checks will be as we manipulate the values here.

## How Has This Been Tested?

Interactively only

## Where Has This Been Documented?

Docstrings only

## Screenshots (if appropriate):
First open:
image

Sensible settings:
image

The borders move:
image

Inclusive:
image

Inversion:
image

### CODE REVIEW

**Comment by ZLLentz** [comment]
There are some sizing issues if the user puts in a really really big number- not sure how important these would be

**Comment by ZLLentz** [comment]
I'm also now realizing before I skip off to a late lunch- inversion coloring might not be correct for the warning ranges, I'm not 100% sure of the implementation

**Comment by klauer** [comment]
> There are some sizing issues if the user puts in a really really big number- not sure how important these would be

Not important, in my opinion.

> I'm also now realizing before I skip off to a late lunch- inversion coloring might not be correct for the warning ranges, I'm not 100% sure of the implementation

Inverting with warning ranges is a bit too confusing, if you ask me. Ignoring the inclusive/exclusive setting:
With non-inverted logic:
  * No warning limits set
    * If below `low` or above `high` -> `severity_on_failure`
    * Value falling within `low`/`high` -> success
  * Warning limits set
    * If below `low` or above `high` -> `severity_on_failure`
    * Warn if value is in `(low, warn_low)` or `(warn_high, high)`. 
    * Value falling within `warn_low`/`warn_high` -> success


With inverted logic:
  * No warning limits set
    * If value between `low` and `high` -> `severity_on_failure`
    * Value falls outside `low` and `high` -> success
  * Warning limits set
     * Invert the logic above for each step; I don't think it's sensible

**Review by klauer — APPROVED** [review]
I think this looks really great and it should be rather intuitive for the end-user. I plan on playing around with this more in a bit. That shouldn't hold up merging this PR.

File: atef/widgets/config.py [hunk 1]
```diff
@@ -2210,3 +2213,220 @@ class LessOrEqualWidget(CompMixin, GtLtBaseWidget):
...
+        """
+        # Cute trick: swap red and green if we're inverted
+        if self.bridge.invert.get():
+            green = self.red
+            red = self.green
+        else:
+            green = self.green
+            red = self.red
+        yellow = self.yellow
+        self.left_red_line.penColor = red
```
Comment (klauer) [inline atef/widgets/config.py:2359]: I think the vertical lines could use a larger pen width to make them more visible

Comment (ZLLentz) [reply atef/widgets/config.py:2359]: Easy to adjust

**Comment by ZLLentz** [comment]
> Inverting with warning ranges is a bit too confusing

I think I agree- should I just hide the warning thresholds under inversion?

File: atef/widgets/config.py (hunk 1 above)
Comment (ZLLentz) [reply atef/widgets/config.py:2359]: Thicker lines: 
image

**Comment by ZLLentz** [comment]
I'm going to remove the yellow region under inversion, much like how the gui displays with no warning threshold set, to make it clear that "fail with warning" is not a sensible outcome for a test

**Pull request merged**